lib/posix-process/clone.c

```diff
@@ -388,6 +389,27 @@ static int _clone(struct clone_args *cl_args, size_t cl_args_len,
 	uk_pr_debug(")\n");
 #endif /* UK_DEBUG */
 
+	/* CLONE_VM requires that the child operates on the same memory
+	 * space as the parent.
+	 *
+	 * Assign the parent's stack. We assign the user TLS to the
+	 * parent's in clone_setup_arch_sysregs(). We copy the registers
+	 * later below.
+	 */
+	if ((flags & CLONE_VM)) {
+		if (!cl_args->stack && !cl_args->stack_size) {
+			cl_args->stack = ukarch_regs_get_sp(&execenv->regs);
```

Comment:
Yes, I went through that path as well. The problem with using local variables is that it would require making every subsequent access to `cl_args->stack` and `cl_args->tls` conditional to:
```c
if ((flags & CLONE_VM) && (!cl_args->stack))
  /* use local var */
else
 /* use cl_args->stack */

if ((flags & CLONE_VM) && (!cl_args->tls))
  /* use local var */
else
 /* use cl_args->tls */
```
That would result in a lot of code duplication which I wanted to avoid as the implementation of clone is complex enough already.

Reply:
ok but this still feels awkward and I am thinking of the case where someone may pass something to clone and wake up to clonetab having a different value somehow. Maybe at least there should be a warning/note comment somewhere?